Assets/MixedRealityToolkit/UX/Scripts/Progress/FadeObject.cs

```diff
@@ -100,19 +100,28 @@ void Update()
                 {
                     percent = 1 - percent;
                     if (percent < mCachedColor.a)
+                    {
                         mCachedColor.a = percent;
+                    }
                 }
                 else
                 {
                     if (percent > mCachedColor.a)
+                    {
                         mCachedColor.a = percent;
+                    }
                 }
 
-                if (mRenderer != null)
+                if (mCachedMaterial != null)
                 {
-                    mRenderer.material.color = mCachedColor;
+                    mCachedMaterial.color = mCachedColor;
                 }
             }
         }
+
+        public void OnDestroy()
+        {
+            DestroyImmediate(mCachedMaterial);
```

Comment:
`DestroyImmediate` should only be used for editor code only.  Please use `Destroy` instead.